File: src/main/java/com/example/progettoenterprise/dto/AmiciziaDTO.java
package com.example.progettoenterprise.dto;

import com.example.progettoenterprise.data.entities.Amicizia.StatoAmicizia;
import lombok.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AmiciziaDTO {

    private Long id;

    private Long richiedenteId;
    private String richiedenteUsername;

    private Long riceventeId;
    private String riceventeUsername;

    private StatoAmicizia stato;
    private LocalDateTime dataRichiesta;
    private LocalDateTime dataRisposta;
}

File: frontendMobile/app/src/main/java/com/example/enterprisemobile/model/dto/AmiciziaDTO.kt
package com.example.enterprisemobile.data.model

data class AmiciziaDTO(
    val id: Long,
    val richiedenteId: Long,
    val richiedenteUsername: String,
    val riceventeId: Long,
    val riceventeUsername: String,
    val stato: String,
    val dataRichiesta: String? = null,
    val dataRisposta: String? = null
)